### ISSUE TITLE
Missing RJMCMC (reversible-jump) sampler

### ISSUE THREAD

**Issue by Moelf** [TuringLang/Turing.jl#2023]
Just wanna open an issue to track this

**Comment by Red-Portal** [comment]
What do you mean by track? Was there any plan to include an RJMCMC sampler at any point?

**Comment by Moelf** [comment]
no, but someone may also want this and this can track how many people may want it

**Comment by Red-Portal** [comment]
As long as you can infer which variable is the model indicator and the dimensionality of the parameters can change, one could, in principle, implement such a backend for Turing. I think I vaguely remember that DynamicPPL cannot handle dynamically changing dimensions, but maybe I'm wrong on this. 

The thing about RJMCMC that always concerns me is that RJMCMC tries to estimate Bayes factors for the Metropolis-Hastings step. And since estimating Bayes factors is rarely a good idea, the cases where RJMCMC actually works are pretty limited. They are notoriously hard to diagnose, too. They might look like they are working, but it's hard to tell if it's not producing garbage.

**Comment by penelopeysm** [comment]
FWIW, at a tutorial we did this week (https://www.cmstatistics.org/CFECMStatistics2025/tutorials.php) somebody asked about reversible jump MCMC was available, so there is definitely some interest.

**Comment by penelopeysm** [comment]
Don't hold your breath though, I doubt it will happen in the near future.

**Comment by Red-Portal** [comment]
@penelopeysm I do have a working, tested, and fairly general implementation under my [sleeve](https://github.com/Red-Portal/reversibleJump.jl). I am planning to work on it once `AdvancedVI` becomes more mature and hopefully make it compatible with Turing. So maybe in a year or two we'll have something?

**Comment by penelopeysm** [comment]
Ooh! Yes definitely, let's talk about it next year.
